docs/code-docs/source/zero3.rst

```diff
@@ -293,3 +293,37 @@ parallelism to fit them in limited GPU memory.
 
 .. autoclass:: deepspeed.zero.TiledLinear
     :members:
+
+
+Debugging
+---------
+
+Debugging ZeRO training is complicated by the partitioning of parameters, gradients, and optimizer states.
+To simplify debugging, DeepSpeed provides the following routines for accessing individual model states
+in their full form (i.e., un-partitioned).
```

Comment:
```suggestion
Debugging ZeRO training is complicated by the partitioning of parameters, gradients, and optimizer states.
None of these 3 groups of parameters can be normally accessed because of that. To overcome that DeepSpeed provides the following utilities for accessing individual model states in their unpartitioned form.

Please note that these utilities must be called by all processes, even if you decide to do something with the result only in the main process. If all processes don't participate these utilities will hang waiting for all processes to send their contribution.
```

Reply:
please review/extend my additions on when the data is valid.